### ISSUE TITLE
Request for stable release (=version 1.0)

### ISSUE THREAD

**Issue by 304NotModified** [AngleSharp/AngleSharp#709]
Hi @FlorianRappl 

First of all, thanks for this great library! I think this library is the best in it's category :)

Unfortunately, it's a bit pity there is no stable (1.0) version released of this library. For me it's a bit frustrating as I expected a stable a few years ago. While there is some development now after years, I'm afraid there won't be a stable release ever. 

I understand if you're busy with other things in life and sometimes things won't go as planned. But thinking out-of-the-box, I think the open issues for 1.0 is to huge. 

Also the breaking changes are a bit disturbing, as libraries build on this library has a hard time  to get stable. (for example the excellent library HtmlSanitizer - see [issues](https://github.com/mganss/HtmlSanitizer/issues?q=is%3Aissue+anglesharp+in%3Atitle+is%3Aclosed))

So I have a proposal:

- Skip all issues from [v0.10](https://github.com/AngleSharp/AngleSharp/milestone/25) that aren't bugs. Those aren't critical. Sure, better performance is better, build on Linux is cool, but it's isn't really necessary for a 1.0
- Skip [v0.11](https://github.com/AngleSharp/AngleSharp/milestone/28). I think that would be nice for 1.1. Or maybe 2.0
- Release 1.0 alphas/betas soon. An alpha could be released already with the 4 open bugs in v0.10
- Last but not least (not sure if it was planned already), please adopt semver (= no breaking changes in 1.x)

Cheers,
Julian

PS: I apologize if I sound blunt. Direct language is very common in my home country 👼

**Comment by FlorianRappl** [comment]
The reason for 0.10 is to get feedback for the API and have a stable 1.0. And yes, of course semver is used for 1.0 onwards (I think this is even described in one of the docs).

However, most bullet points still assume that some development is done. The truth is as simple as bitter: The .NET ecosystem and its community did not keep my motivation for development in this space. I still want to conclude the work on this one (i.e., at least leave a 1.0 version behind where only some hotfixes / improvements should be merged in from time-to-time), but time is an issue.

Nevertheless, even though I cannot promise anything I still have AngleSharp on the radar and I'll try to deliver hopefully more soon-ish than late / never-ish. Contributions in any space are welcome. One of the greatest killers was Microsoft's ever growing change. Once we had the nice multi-platform build using dotnet with project.json established the system was outdated. Then NuGet went on to abandon multi-platform support and it became hard to make the system build again on AppVeyor. Hence I believe that a 0.10 will drop most of the multi-platform to only go with .NET Standard (that is of course implied multi-platform, however, not containing all earlier supported platforms).

Regarding 0.11 - I inserted this one for some modifications prior to 1.0 that still should validated before going semver and having to release 2.0 instantly.

TL;DR: The proposal is of course nice and goes in line with my thoughts, however, zero progress is still zero progress no matter how much the scope is reduced.

HTH and thanks for your support! :beers:

**Comment by 304NotModified** [comment]
Thanks for the fast response! 👍 

I fully understand your situation and therefore I try to get a stable, with the minimal effort needed. 



> One of the greatest killers was Microsoft's ever growing change.

Luckily I think it's stable now :) Maybe not fully mature, but stable. 

> Regarding 0.11 - I inserted this one for some modifications prior to 1.0 that still should validated before going semver and having to release 2.0 instantly.

Do you think "HTML 5.2" (#618) would introduce breaking changes? (PS #535 could be fixed easily without breaking changes)

I don't wanna repeat myself, but there are barely issues in the milestones 0.10, 0.11 and 1.0 that should block a stable release.

> And yes, of course semver is used for 1.0 onward

👍  great news! As a developer, I really like semver

**Comment by FlorianRappl** [comment]
> Do you think "HTML 5.2" (#618) would introduce breaking changes?

No the 0.11 is more a "guess" than a fixed set. I just placed also some other things (e.g., #618) in this milestone, but nothing cut in stone and such issues could easily be moved if there is no other reason for a 0.11.

**Comment by 304NotModified** [comment]
OK, but why before 1.0? 1.1 sounds reasonable to me.

**Comment by sakshamsaxena** [comment]
Hey @FlorianRappl 
Is there a roadmap yet to v1 ? I read the above comments and can fully sympathize. I'd like to contribute the best I can to reach v1 ! 😄 

(sorry if this is the wrong place for ice-breaker 😮 )

**Comment by FlorianRappl** [comment]
Hi @sakshamsaxena - thanks for bringing this up. Current plan is to release 0.11 today (it will - among others - feature documentation improvements, some bug fixes and further stabilization, exposing more possibilities, and removes AngleSharp.Xml, which will be available in a dedicated NuGet package).

If everything goes well I would assume 1.0 to hit production eo Q1 / so Q2 this year. API-wise I think we are good, I only want to make sure there are no blockers / hard bugs in there, which may require an API change or other breaking things.

HTH!

(any help welcome - especially in the areas of perf improvements, documentation, eco-system, ... - if you want to contribute I would more than appreciate it; just tell me your preferences and we are good to go!)

**Comment by sakshamsaxena** [comment]
Can help in perf definitely ! Also, good to hear on these updates ❤️

**Comment by FlorianRappl** [comment]
We will ship a preview for v1 later today.

Depending on the incoming bug requests and the backlog we'll ship v1 end of April / in May.

**Comment by 304NotModified** [comment]
What happened to v1?

**Comment by 304NotModified** [comment]
@FlorianRappl bump. Any idea if v1 will be this year? 

This request is many years old.

**Comment by FlorianRappl** [comment]
@304NotModified happens when the documentation is up to the standard. If you want to accelerate it feel free to contribute!
